docs/user_guide/security/identity_security.rst

```diff
@@ -0,0 +1,503 @@
+                "view": "any",  # can do commands in the "view" category
+                "operate": "none"  # cannot do commands in "operate" category
+            }
+        }
+    }
+
+.. _site_specific_auth:
+
+Site-specific Authentication and Federated Job-level Authorization
+==================================================================
+Site-specific authentication and authorization allows the customers to inject their own authentication and
+authorization methods into the NVFlare system. This includes the FL server / clients registration, authentication,
+and the job deployment and run authorization.
+
+NVFlare provides a general purpose event based pluggable authentication and authorization framework to allow customers to expand functionality such as:
+
+    - exposing the app through a WAF (Web Application Firewall) or any other network element enforcing MTLS
+    - using a confidential certification authority to ensure the identity of each participating site and to ensure that they meet the computing requirements for confidential computing
+    - defining additional roles to manage who can submit which kind of jobs to execute within NVFlare, identify who submits jobs and which dataset can be accessed
+
+Customers can write their own :ref:`FLComponents <fl_component>`, listening to the NVFlare system events at different points of their workflow,
+then easily plug in their authentication and authorization logic as needed.
+
+Assumptions and Risks
+---------------------
+By enabling the customized site-specific authentication and authorization, NVFlare will make several security
+related data available to the external FL components, e.g. IDENTITY_NAME, PUBLIC_KEY, CERTIFICATE, etc. In order
+to protect them from being compromised, that data needs to be made read-only.
+
+Because of the external pluginable authentication and authorization processes, the results of the processes could
+potentially cause the jobs to not be able to be deployed or run. When configuring and using these functions, the customers
+need to be aware of the impact and know where to plug in the authentication and authorization check.
+
+Event based pluginable authentication and authorization
+-------------------------------------------------------
+We will use the NVFlare event based solution to support the site-specific authentication and federated job-level authorization
+from the NVFlare platform. We will identify the proper points when each individual sites about the start, register and connect
+to the server, admins upload and submit the jobs to the system, the job started to dispatch to the sites, etc, fire the
+specific system events which allow the users to build and plugin their FLcomponents, listen to the events and provide their
+own authentication and authorization functions. In this case, you will be able to provide and implement any sort of
+additional security checks. 
+
+.. code-block:: python
+
+    class EventType(object):
```

Comment:
it will be better if we could provide a pseudo-loop or something that tells people when these events are fired.

Reply:
This can be enhancement follow up PR @nvkevlu